src/DGtal/shapes/MeshVoxelizer.ih

```diff
@@ -0,0 +1,241 @@
+  else if(val1 < 0. || val2 < 0. || val3 < 0.)
+    return OUTSIDE;
+  else if(val1 == 0. || val2 == 0. || val3 == 0.)
+    return ONEDGE;
+  else
+    return INSIDE;
+}
+
+// ---------------------------------------------------------
+template <typename TDigitalSet, int Separation>
+inline
+bool
+DGtal::MeshVoxelizer<TDigitalSet, Separation>::pointIsInsideVoxel(const PointR3& P,
+                                                                  const PointZ3& v)
+{
+  bool isInside = true;
+
+  PointR3 PP = P - v;
+  for(int i(0); i < 3; i++)
+    isInside = isInside && ( -0.5 <= PP[i] && PP[i] <= 0.5 );
+
+  return isInside;
+}
+
+// ---------------------------------------------------------
+template <typename TDigitalSet, int Separation>
+inline
+void
+DGtal::MeshVoxelizer<TDigitalSet, Separation>::voxelizeTriangle(DigitalSet &outputSet,
+                                                                const PointR3& A,
+                                                                const PointR3& B,
+                                                                const PointR3& C,
+                                                                const VectorR3& n,
+                                                                const std::pair<PointR3, PointR3>& bbox)
+{
+  OrientationFunctor orientationFunctor;
+
+  //geometric predicate
+  PredicateFromOrientationFunctor2<OrientationFunctor> pointPredicate( orientationFunctor );
+
+  // foreach intersection target
+  for(unsigned int i(0); i < myIntersectionTarget().size(); i++)
+  {
+    // 2D projection of A ; B ; C
+    PointR2 AA = myIntersectionTarget.project(i, A);
+    PointR2 BB = myIntersectionTarget.project(i, B);
+    PointR2 CC = myIntersectionTarget.project(i, C);
+
+    // check orientation
+    if(! pointPredicate(AA, BB, CC))
+      std::swap(AA, CC);
+
+    // traverse all voxel of current face bounding box
+    PointZ3 v = bbox.first;
+    for(; v[1] <= bbox.second[1]; v[1]++)
+      for(v[0] = bbox.first[0]; v[0] <= bbox.second[0]; v[0]++)
+        for(v[2] = bbox.first[2]; v[2] <= bbox.second[2]; v[2]++)
+        {
+          auto target = myIntersectionTarget(i);
+          // check if points are on different side
+          target.myFirst += v;
+          target.mySecond += v;
+
+          VectorR3 a2myFirst = A - target.myFirst;
+          VectorR3 a2mySecond = A - target.mySecond;
+
+          VectorR3 w = target.mySecond - target.myFirst;
+          double den = n.dot(w);
+
+          bool isSameSide = den == 0 // target on plane
+            || ( a2myFirst.dot(n) * a2mySecond.dot(n) > 0 ); // target on one side
+
+          // if target is on the same side -> skip current iteration
+          if( isSameSide )
+            continue;
+
+          PointR2 pp = myIntersectionTarget.project(i, v);
+
+          // check if current voxel projection is inside ABC projection
+          if(pointIsInside2DTriangle(AA, BB, CC, pp) != OUTSIDE)
+            outputSet.insert(v);
```

Comment:
here I got an error if the scaleFactor is too large (v is outside the output domain, see previous remarks).